src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/integration/SetupAssistant.kt

```diff
@@ -215,17 +215,34 @@ class SetupAssistant(
         sentBy = ppUser
       )
     )
-    referral.supplierAssessment = createSupplierAssessment(referral = referral, appointments = mutableSetOf())
+    referral.supplierAssessment = createSupplierAssessment(referral = referral)
     return referral
   }
 
   fun addSupplierAssessmentAppointment(
     supplierAssessment: SupplierAssessment,
-    appointment: Appointment = appointmentFactory.create(createdBy = createPPUser())
+    appointment: Appointment = appointmentFactory.create(createdBy = createPPUser()),
+    appointmentDeliveryType: AppointmentDeliveryType? = null,
+    appointmentDeliveryAddress: AddressDTO? = null
   ) {
     appointmentRepository.save(appointment)
     supplierAssessment.appointments.add(appointment)
-    supplierAssessmentRepository.save(supplierAssessment)
+    supplierAssessmentRepository.saveAndFlush(supplierAssessment)
+    if (appointmentDeliveryType != null) {
```

Comment:
_nit_ `appointmentDeliveryType` is never null in the test usage, so this branch could be skipped

Reply:
I've removed this